### PR TITLE
Fix deprecation issues for PHP 8.1

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Fix deprecation warnings for PHP 8.1. compatibility.

### Related Issues
1. magento/magento2#34567

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/Stdlib/StringUtils.php [hunk 1]
```diff
@@ -149,7 +149,7 @@ public function split($value, $length = 1, $keepWords = false, $trim = false, $w
      */
     public function strlen($string)
     {
-        return mb_strlen($string, self::ICONV_CHARSET);
+        return $string !== null ? mb_strlen($string, self::ICONV_CHARSET) : '';
```
Comment (andrewbess) [inline lib/internal/Magento/Framework/Stdlib/StringUtils.php]: Hello @karyna-tsymbal-atwix.
I think this is a typo.
Please, fix it

```suggestion
        return $string !== null ? mb_strlen($string, self::ICONV_CHARSET) : 0;
```

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @karyna-tsymbal-atwix 
Thank you for your contribution.
Please check my comment

**Comment by andrewbess** [comment]
@magento run all tests with env PHP 8.1

**Comment by andrewbess** [comment]
@magento run all tests

**Review by andrewbess — APPROVED** [review]
Fixes look good for me

**Comment by xmav** [comment]
pending delivery https://github.com/magento-commerce/magento2ce/pull/7322

**Pull request merged**